### PR TITLE
[Translation] Ability to format a message catalogue without actually writing it.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | #5743 |

### CODE REVIEW

**Comment by xabbuh** [comment]
Can we add a `versionadded` directive for the `formatCatalogue()` method?

**Comment by aitboudad** [comment]
done

File: components/translation/custom_formats.rst [hunk 1]
```diff
@@ -117,3 +117,9 @@ YAML file are dumped into a text file with the custom format::
     $dumper = new MyFormatDumper();
     $dumper->dump($catalogue, array('path' => __DIR__.'/dumps'));
 
+.. versionadded:: 2.8
+    The ability to format a message catalogue without dumping it was introduced in Symfony 2.8.
+
+.. note::
```
Comment (xabbuh) [inline components/translation/custom_formats.rst]: I think this should be a tip and we should possibly move it somewhere up.

Comment (aitboudad) [inline components/translation/custom_formats.rst]: updated to tip but I couldn't find where I can move it!

File: components/translation/custom_formats.rst [hunk 2]
```diff
@@ -117,3 +117,9 @@ YAML file are dumped into a text file with the custom format::
     $dumper = new MyFormatDumper();
     $dumper->dump($catalogue, array('path' => __DIR__.'/dumps'));
 
+.. versionadded:: 2.8
+    The ability to format a message catalogue without dumping it was introduced in Symfony 2.8.
+
+.. tip::
+
+    To format a message catalogue without dumping it into files use the `formatCatalogue` method.
```
Comment (WouterJ) [inline components/translation/custom_formats.rst]: you have to use double  backticks to indicate inline code blocks in reStructured Text

Comment (aitboudad) [inline components/translation/custom_formats.rst]: fixed

File: components/translation/custom_formats.rst (hunk 1 above)
Comment (xabbuh) [inline components/translation/custom_formats.rst]: Maybe moving the `versionadded` directive and the tip above the second code block in this section?

Comment (aitboudad) [inline components/translation/custom_formats.rst]: done

**Comment by xabbuh** [comment]
Hm, actually I am not very satisfied with this approach too. What if tried a totally different approach and add a sidebar that is talking about the `formatCatalgoue()` method and also contains this tip?

**Comment by aitboudad** [comment]
@xabbuh can you look now ?

**Comment by xabbuh** [comment]
:+1: Yeah, I like that much more. :)

File: components/translation/custom_formats.rst [hunk 3]
```diff
@@ -102,7 +102,16 @@ will save a few lines::
         }
     }
 
-The :method:`Symfony\\Component\\Translation\\Dumper\\FileDumper::format`
+.. sidebar:: Format a message catalogue
+
+    .. versionadded:: 2.8
+        The ability to format a message catalogue without dumping it was introduced in Symfony 2.8.
+
+    In some cases, you want to send the the dump contents as a response instead of writing in files.
```
Comment (javiereguiluz) [inline components/translation/custom_formats.rst]: `send the the dump` -> `send the dump`

Comment (javiereguiluz) [inline components/translation/custom_formats.rst]: `instead of writing in files` -> `instead of writing them in files`

Comment (aitboudad) [inline components/translation/custom_formats.rst]: fixed

**Comment by javiereguiluz** [comment]
:+1: thanks @aitboudad

**Comment by xabbuh** [comment]
Thank you @aitboudad.

**Pull request merged**